migrations/Version202409231414061488_taoAdvancedSearch.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use oat\generis\model\DependencyInjection\ServiceOptions;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use oat\taoAdvancedSearch\model\SearchEngine\Driver\Elasticsearch\ElasticSearchConfig;
+use oat\taoAdvancedSearch\model\SearchEngine\Driver\Elasticsearch\ElasticSearchEnvConfig;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ *
+ * phpcs:disable Squiz.Classes.ValidClassName
+ */
+final class Version202409231414061488_taoAdvancedSearch extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Update ElasticSearchConfig to use properties from environment variables';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $serviceManager = $this->getServiceManager();
+        $serviceOptions = $this->getServiceOptions();
+        $options = $serviceOptions->getOptions();
+
+        if (
+            (
+            $options[ElasticSearchConfig::class][ElasticSearchConfig::OPTION_ELASTIC_CLOUD_ID]
```

Comment:
Direct access to not existing array attributes will generate warning. We need to verify that the array index exists, first